core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -94,6 +94,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   // contains a map from hostname to a list of input format splits on the host.
   private[spark] var preferredNodeLocationData: Map[String, Set[SplitInfo]] = Map()
 
+  // This is used for Spark Streaming to check whether driver host and port are set by user,
+  // if these two configurations are set by user, so the recovery mechanism should not remove this.
+  private[spark] val isDriverHostSetByUser = config.contains("spark.driver.host")
```

Comment:
It doesn't seem worth tacking on yet more little fields in `SparkContext` just for a niche use case in a submodule. Use the config object in `Checkpoint`.